allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_lims_api.py

```diff
@@ -106,20 +106,94 @@ def get_nwb_filepath(self) -> str:
     def get_eye_tracking_filepath(self) -> str:
         """Get the filepath of the eye tracking file (*.h5) associated with the
         ophys experiment"""
-        query = """
-                SELECT wkf.storage_directory || wkf.filename
-                AS eye_tracking_file
+        query = f"""
+                SELECT wkf.storage_directory || wkf.filename AS eye_tracking_file
                 FROM ophys_experiments oe
-                LEFT JOIN well_known_files wkf
-                ON wkf.attachable_id = oe.ophys_session_id
-                JOIN well_known_file_types wkft
-                ON wkf.well_known_file_type_id = wkft.id
+                LEFT JOIN well_known_files wkf ON wkf.attachable_id = oe.ophys_session_id
+                JOIN well_known_file_types wkft ON wkf.well_known_file_type_id = wkft.id
                 WHERE wkf.attachable_type = 'OphysSession'
-                AND wkft.name = 'EyeTracking Ellipses'
-                AND oe.id = {};
-                """.format(self.get_ophys_experiment_id())
+                    AND wkft.name = 'EyeTracking Ellipses'
+                    AND oe.id = {self.get_ophys_experiment_id()};
+                """
+        return safe_system_path(self.lims_db.fetchone(query, strict=True))
+
+    @memoize
+    def get_eye_gaze_mapping_file_path(self) -> str:
+        """Get the filepath of the eye gaze mapping file (*.h5) associated with the
+        ophys experiment"""
+        query = f"""
+                SELECT wkf.storage_directory || wkf.filename AS eye_tracking_file
+                FROM ophys_experiments oe
+                LEFT JOIN well_known_files wkf ON wkf.attachable_id = oe.ophys_session_id
+                JOIN well_known_file_types wkft ON wkf.well_known_file_type_id = wkft.id
+                WHERE wkf.attachable_type = 'OphysSession'
+                    AND wkft.name = 'EyeDlcScreenMapping'
+                    AND oe.id = {self.get_ophys_experiment_id()};
+                """
         return safe_system_path(self.lims_db.fetchone(query, strict=True))
 
+    @memoize
+    def get_eye_tracking_rig_geometry(self) -> Optional[dict]:
+        """Get the eye tracking rig geometry metadata"""
+        ophys_experiment_id = self.get_ophys_experiment_id()
+
+        query = f'''
+            SELECT oec.*, oect.name as config_type, equipment.name as equipment_name
+            FROM ophys_sessions os
+            JOIN observatory_experiment_configs oec ON oec.equipment_id = os.equipment_id
+            JOIN observatory_experiment_config_types oect ON oect.id = oec.observatory_experiment_config_type_id
+            JOIN ophys_experiments oe ON oe.ophys_session_id = os.id
+            JOIN equipment ON equipment.id = oec.equipment_id
+            WHERE oe.id = {ophys_experiment_id} AND oect.name IN ('eye camera position', 'led position', 
+                'screen position')
+        '''
+        api = (credential_injector(LIMS_DB_CREDENTIAL_MAP)
```

Comment:
I think this can be replaced by calling `self.lims_db`? So no need to create another connection I think?

Reply:
I think you are right. I mistakenly copied some other code from this module which used this approach. I will fix it.